packages/launchdarkly-adapter/modules/adapter/adapter.js

```diff
@@ -1,25 +1,56 @@
+// @flow
+
 import { initialize } from 'ldclient-js';
 import camelCase from 'lodash.camelcase';
 
-const adapterState = {
+type FlagValue = boolean | string;
+type FlagName = string;
+type User = {
+  key: string,
+};
+type Client = {
+  identify: (nextUser: User) => void,
+  on: (state: string, () => void) => void,
+  on: (state: string, (flagName: FlagName) => void) => void,
+  allFlags: () => Flags | null,
+};
+type Flag = [FlagName, FlagValue];
+type Flags = { [FlagName]: FlagValue };
+type OnFlagsStateChangeCallback = Flags => void;
+type OnStatusStateChangeCallback = ({ [string]: boolean }) => void;
+
+const adapterState: {
+  isReady: boolean,
+  isConfigured: boolean,
+  user: ?User,
+  client: ?Client,
+} = {
```

Comment:
What about an `AdapterState` type?

Reply:
Yup, had that and struggled a bit with it. Especially as it's not reused. To me, right now, types make sense when you use them in generics for e.g. React.Component.

Reply:
Sure, I think often it comes down to how much you re-use that type.

Reply:
a58d575